src/translator/cache.cpp

```diff
@@ -0,0 +1,157 @@
+#include "translator/cache.h"
+
+#include <cstdlib>
+
+namespace marian {
+namespace bergamot {
+
+#ifndef WASM_COMPATIBLE_SOURCE
+
+ThreadSafeL4Cache::ThreadSafeL4Cache(const CacheConfig &config)
+    : epochManagerConfig_(config.ebrQueueSize, std::chrono::milliseconds(config.ebrIntervalInMilliseconds),
+                          config.ebrNumQueues),
+      cacheConfig_(config.size * 1024 * 1024, std::chrono::seconds(config.timeToLiveInMilliseconds),
+                   config.removeExpired),
+      service_(epochManagerConfig_),
+      context_(service_.GetContext()) {
+  // There is only a single cache we use. However, L4 construction API given by example demands we give it a string
+  // identifier.
+  const std::string cacheIdentifier = "global-cache";
+  hashTableIndex_ = service_.AddHashTable(L4::HashTableConfig(
+      cacheIdentifier, L4::HashTableConfig::Setting{static_cast<uint32_t>(config.numBuckets)}, cacheConfig_));
+}
+
+bool ThreadSafeL4Cache::fetch(const marian::Words &words, ProcessedRequestSentence &processedRequestSentence) {
+  auto &hashTable = context_[hashTableIndex_];
+
+  /// We take marian's default hash function, which is templated for hash-type. We treat marian::Word (which are indices
+  /// represented by size_t (unverified) and convert it into a uint64_t for hashing with more bits)
+  ///
+  /// Note that there can still be collisions, in which case the translations returned from cache can be wrong. It's
+  /// practically very unlikely (claims @XapaJIaMnu). To guarantee correctness, comparisons need to be made with the
+  /// actual key, which is cannot be done with this approach, as the cache is unaware of the original marian::Words key.
+  //
+  /// TODO(@jerinphilip): Empirical evaluation that this is okay.
+
+  KeyBytes keyBytes;
+  auto key = hashFn_(words);
```

Comment:
uint64_t pls

Reply:
All functions are reliant on std::hash, which returns `std::size_t` [per cppreference](https://en.cppreference.com/w/cpp/utility/hash). I have updated things to size_t instead.